WaveInt.py

```diff
@@ -208,7 +208,8 @@ def calcKernel(self,ofd=sys.stdout,efd=sys.stderr):
         '''
         
         # Check/convert source coordinates
-        self.checkXs()
+        if calc_dist ==False:
```

Comment:
self.checkXs does not compute any distance, it is just checking if self.Xs has been assigned correctly. Why would you want to skip this part?

Reply:
There is a bug in the workflow when using `synthSDR` because we check twice self.Xs. It goes like this:

1. `synthSDR` call the `calcDist` function.
2. `calcDist` function calls `checkXs`.
3. At the end of `checkXs`, self.Xs  is an array.
4. Then `synthSDR` calls the `calcKernel` function.
5.  `calcKernel` checks again with `checkXs`.
6. `checkXs` already transformed self.Xs  to an array. So when you asset in line 129 `assert self.Xs!=None,`  it will ask to include either .any() or .all() because it is already an array.

I hope it was clearer.

Reply:
Ok thanks for your quick reply. I just modified the `checkXs` method to correct this. 
`calcDist` and `calcKernel` can be called independently so I preferred to keep the calls to `checkXs` in `calcDist` and `calcKernel` to check the format `self.Xs`. 
With the latest version of the master branch, the bug should not be here anymore and there is no need to add an extra parameter to the `calcDist` and `calcKernel` methods.

Please test the latest version of the master branch and let me know if it works.